src/System.Management.Automation/engine/hostifaces/PSTask.cs

```diff
@@ -428,6 +438,19 @@ public void Start()
             _runspace.Name = string.Format(CultureInfo.InvariantCulture, "{0}:{1}", RunspaceName, s_taskId);
             _runspace.Open();
 
+            // Set current working directory on the runspace to be the same as the calling script.
+            // Temporarily set the newly created runspace as the thread default runspace for any needed module loading.
+            var oldDefaultRunspace = Runspace.DefaultRunspace;
+            try
+            {
+                Runspace.DefaultRunspace = _runspace;
+                _runspace.ExecutionContext.SessionState.Internal.SetLocation(_currentLocation.Path);
```

Comment:
It seems we don't check the currentLocation parameter to null in public API.

Reply:
Yeah, it does look like the 'CurrentDrive' can be null, and the current location not defined.  I'll add code to handle that.